src/core/context/context-api.ts

```diff
@@ -296,7 +296,25 @@ export function makeContextApi(ai: ApiInput): EdgeContext {
         stashTree: LoginStash,
         mainStash: LoginStash
       ): Promise<EdgeAccount> {
-        const sessionKey = await loginPin2(ai, stashTree, mainStash, pin, opts)
+        let sessionKey: SessionKey
+        if (stashTree.wipChange != null) {
+          // Try login with the WIP change, in the case where it's the valid
+          // login stash:
+          try {
+            sessionKey = await loginPin2(
+              ai,
+              stashTree.wipChange,
+              mainStash,
+              pin,
+              opts
+            )
+          } catch (error) {
+            // Fail gracefully with the original stash tree...
+            sessionKey = await loginPin2(ai, stashTree, mainStash, pin, opts)
+          }
```

Comment:
Optional: This might read cleaner as `sessionKey = await loginPin2(...wip...).catch(() => loginPin2(...mainStash...)`, but your way works too.